extensions/frontend-tiles/src/FrontendTiles.ts

```diff
@@ -3,22 +3,133 @@
  * The tileset is expected to reside at "baseUrl/tileset.json" and to have been produced by the [mesh export service](https://developer.bentley.com/apis/mesh-export/).
  * If no such tileset exists for the given iModel, return `undefined`.
  * @see [[FrontendTilesOptions.computeSpatialTilesetBaseUrl]].
- * @alpha
+ * @beta
  */
 export type ComputeSpatialTilesetBaseUrl = (iModel: IModelConnection) => Promise<URL | undefined>;
 
+/** @internal exported for tests. */
+export function createMeshExportServiceQueryUrl(args: ObtainTilesetUrlFromMeshExportServiceArgs): string {
+  const prefix = args.urlPrefix ?? "";
+  let url = `https://${prefix}api.bentley.com/mesh-export/?iModelId=${args.iModel.iModelId}`;
+  if (!args.noSort)
+    url = `${url}&$orderBy=date:desc`;
+
+  return url;
+}
+
+/** Represents the result of a [mesh export](https://developer.bentley.com/apis/mesh-export/operations/get-export/#export).
+ * @see [[queryCompletedMeshExports]].
+ * @beta
+ */
+export interface MeshExport {
+  id: string;
+  displayName: string;
+  status: string;
+  request: {
+    iModelId: string;
+    changesetId: string;
+    exportType: string;
+    geometryOptions: any;
+    viewDefinitionFilter: any;
+  };
+  /* eslint-disable-next-line @typescript-eslint/naming-convention */
+  _links: {
+    mesh: {
+      href: string;
+    };
+  };
+}
+
+/** Arguments supplied to [[queryCompletedMeshExports]].
+ * @beta
+ */
+export interface QueryCompletedMeshExportsArgs {
+  /** The iModel whose exports to query. */
+  iModel: IModelConnection;
+  /** Token used to access the mesh export service.
+   * If omitted, [IModelApp.getAccessToken]($frontend) is used.
+   */
+  accessToken?: AccessToken;
+  /** Disables sorting the results from most to least recent. */
+  noSort?: boolean;
+  /** Optional URL prefix, chiefly useful in testing environments. */
+  urlPrefix?: string;
+}
+
+/** Queries the [mesh export service](https://developer.bentley.com/apis/mesh-export/operations/get-exports/) for completed exports associated with
+ * a specific iModel.
+ * @beta
+ */
+export async function queryCompletedMeshExports(args: QueryCompletedMeshExportsArgs): Promise<MeshExport[]> {
+  let foundExports: MeshExport[] = [];
+
+  const url = new URL(createMeshExportServiceQueryUrl(args));
+  const headers = {
+    /* eslint-disable-next-line @typescript-eslint/naming-convention */
+    Authorization: args.accessToken ?? await IModelApp.getAccessToken(),
+    /* eslint-disable-next-line @typescript-eslint/naming-convention */
+    Accept: "application/vnd.bentley.itwin-platform.v1+json",
+    /* eslint-disable-next-line @typescript-eslint/naming-convention */
+    Prefer: "return=representation",
+  };
+
+  try {
+    const response = await fetch(url, { headers });
+    const result = await response.json() as { exports: MeshExport[] };
+    foundExports = result.exports.filter((x) => x.request.exportType === "IMODEL" && x.status === "Complete");
```

Comment:
Yeah, I briefly had it as `queryMeshExports` with an option to tell it to include incomplete exports, but removed because 1) didn't want this package to become a general-purpose "mesh export service client library" and 2) `obtainTilesetUrlFromMeshExportService` can't do anything useful with an incomplete export.
If I add the option back, will you use it?

Reply:
I amended the API so you can `queryMeshExports` based on your criteria and decide what to do with them (e.g., trigger an export if none exists, identify an in-progress export, etc).